### ISSUE TITLE
Make Gnosis Relay API test use a local Relay service

### ISSUE THREAD

**Issue by nuevoalex** [AugurProject/augur#3129]
Currently the tests for gnosis-relay-api use the Gnosis hosted Rinkeby instance. This makes them slower and less reliable so they currently are not part of CI.

I encountered some issues using the relay locally but in theory it should be possible to spin up a local relay service to handle these tests: https://github.com/gnosis/safe-relay-service

**Comment by pgebheim** [comment]
@robert-e-davidson3 This will come to whence this Epic is scheduled

**Comment by robert-e-davidson3** [comment]
Sounds like we need our CI to spin up a docker for an eth node and a docker for relay, get them working together, then run the relay tests.

Do the relay tests have any node prerequisites that wouldn't be present in our existing geth-pop-15 docker? That may require some work... my guess is we at least need to deploy the Safe contract.

Actually spinning up the docker images and hooking them together is probably easy but @petong would know for sure.

**Comment by nuevoalex** [comment]
FWIW the gnosis safe relay GH has a whole docker based setup as well including support for a ganache instance that it deploys to and all. That's probably the right place to look

**Comment by robert-e-davidson3** [comment]
This is in PR now. My approach in the end was to alter the test to be able to talk to a local relay, including the idiosyncrasy where `blockNumber` isn't returned when talking to ganache, as far as I can deduce. For running the relay itself, I cloned the repo, made a small tweak to its python deps, then built and ran everything with docker-compose.

I updated the `README.md` with these instructions.
